### ISSUE TITLE
RNN Model Architecture

### ISSUE THREAD

**Issue by dxv2k** [VinhDevNguyen/Project_Alpha#4]
## Overview: 
- We can treat each `utrip_id` as a sentence with every word in the sentence is a `city_id` 
- Next, using a sliding window with a length of 4 and feed into the `embedding layer` 
- Loss function can simply understand how its predict next location $$ based on current location
## Model Architecture: 
Contains 3 blocks: `Embedding layer`,  2 layers of `LSTM (1)`, `LSTM (2)`, `Dense with Softmax activation` . 
Picture below provided a better presentation: 
![image](https://user-images.githubusercontent.com/9569076/103457929-e5445c00-4d35-11eb-82fc-6ea4df9a8919.png)

### Embedding layer: 
With such a high density of unique features in the provided dataset. `One-hot encoding` is not a recommendation, I would recommend going with neural-based embedding layer. 

![image](https://user-images.githubusercontent.com/9569076/103458404-3d7d5d00-4d3a-11eb-8825-56ad08c3391b.png)

### LSTM layer: 
LSTM cells: 
![image](https://user-images.githubusercontent.com/9569076/103458488-563a4280-4d3b-11eb-97e8-ce1cfd726f75.png)

### Loss function
![image](https://user-images.githubusercontent.com/9569076/103458468-2854fe00-4d3b-11eb-8119-6f46c89b71f8.png)

## Choosing hyperparameter:
- Embedding layer: 
  * 
- LSTM layer: 
  * `LSTM (1)`
  * `LSTM (2)` 
- Dense with softmax 

Ref: [LSTM-Based Deep Learning Model for Predicting Individual Mobility Traces of Short-Term Foreign Tourists] (https://www.google.com/url?sa=t&rct=j&q=&esrc=s&source=web&cd=&cad=rja&uact=8&ved=2ahUKEwj-sfXIpf3tAhVTlEsFHQ7MCp0QFjABegQIAhAC&url=https%3A%2F%2Fwww.mdpi.com%2F2071-1050%2F12%2F1%2F349&usg=AOvVaw3uurSrALHRK_Wp5GLjS4lD )
